File: GomokuServer/src/GomokuServer.Application/Games/Responses/GetGamesByUsernameResponse.cs
namespace GomokuServer.Application.Games.Responses;

public record GetGamesByUsernameResponse
{
	[Required]
	public required string GameId { get; init; }

	[Required]
	public required UsernamesDto Players { get; init; }

	[Required]
	public bool IsCompleted { get; init; }

	public string? Winner { get; init; }

	[Required]
	public required string Gen { get; init; }

	[Required]
	public required DateTime Date { get; init; }

	public TimeControlDto? TimeControl { get; init; }

	public ClockDto? Clock { get; init; }
}

File: GomokuClient/packages/gomoku-api/client/models/GetGamesByUsernameResponse.ts
import type { UsernamesDto } from "./UsernamesDto";
import type { TimeControlDto } from "./TimeControlDto";
import type { ClockDto } from "./ClockDto";

export type GetGamesByUsernameResponse = {
  /**
   * @type string
   */
  gameId: string;
  /**
   * @type object
   */
  players: UsernamesDto;
  /**
   * @type boolean
   */
  isCompleted: boolean;
  /**
   * @type string
   */
  winner?: string | null;
  /**
   * @type string
   */
  gen: string;
  /**
   * @type string, date-time
   */
  date: Date;
  /**
   * @type object | undefined
   */
  timeControl?: TimeControlDto;
  /**
   * @type object | undefined
   */
  clock?: ClockDto;
};
